### PR TITLE
[yioremote] Fixed reconnection and thing creation issues

### LINKED ISSUE #8593: [yioremote]No reconnection and not possible to create thing
The thing can not be created due to a missmatch of type.xml and YIOremoteBindingConstants.
If the thing is created via *thing file and the connection gets a timeout no reconnection is done

## Expected Behavior
Should be possible to create a thing and if no error is there the reconnection shall work.

## Current Behavior
The thing can not be created due to a missmatch of type.xml and YIOremoteBindingConstants.
If the thing is created via *thing file and the connection gets a timeout no reconnection is done

## Possible Solution
Correction of thingtype.xml and implementing of reconnect function.

## Steps to Reproduce (for Bugs)
1. Add Binding
2. Try to create the thing via paperui

## Your Environment
Current openhab release

### PR DESCRIPTION
**Pull request by miloit**
Fixes #8593

Changed of reconnection issue
Changed of thing-type.xml
Changed readme

### CODE REVIEW

File: bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java [hunk 1]
```diff
@@ -347,6 +366,55 @@ public boolean getAndResetHeartbeat() {
...
+        }
+    }
+
+    public void reconnectWebsocketJob() {
+        switch (yioRemoteDockActualStatus) {
+            case COMMUNICATION_ERROR:
+                logger.debug("Reconnecting YIORemoteHandler");
+                try {
+                    disposeWebsocketPollingJob();
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java:382]: Why `CONFIGURATION_PENDING` here? This is not valid for an `OFFLINE` status.

**Review by kaikreuzer — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java (hunk 1 above)
Comment (miloit) [reply bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java:382]: Which status is than better for reconnection?

Comment (kaikreuzer) [reply bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java:382]: If "reconnection" means that the communication was lost, it should be `OFFLINE.COMMUNICATION_ERRROR`.

Comment (miloit) [reply bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java]: changed

**Review by kaikreuzer — APPROVED** [review]
lgtm, thanks.

**Comment by kaikreuzer** [comment]
cherry-picked to 2.5.x.

**Pull request merged**